JFunc/src/main/java/com/jfunc/validator/FunctionValidator.java

```diff
@@ -25,92 +35,197 @@
-            if (StringUtils.equals(methodName, methodNode.name)) {
-                return getReasons(methodNode, skipLogStatements, skipPrintStatements).toString();
-            }
-        }
-        return "No method found with name" + methodName;
-    }
-
-    // At present we are checking for 1) Void 2) Print-Statements 3) Log-Statements
-    private ObjectNode getReasons(MethodNode methodNode, boolean skipLogStatements, boolean skipPrintStatements) {
-        ObjectMapper mapper = JsonUtils.getObjectMapper();
-        ObjectNode objectNode = mapper.createObjectNode();
-        MethodMetaData methodMetaData = new MethodMetaData(methodNode);
-
-        // First check whether method type is "Void"
-        if (VoidTypeUtil.isVoid(methodMetaData)) {
-            objectNode.put(JfuncConstants.ISFUNCTIONAL, false);
-            objectNode.set(JfuncConstants.REASONS, VoidTypeUtil.getReasons(methodMetaData));
-            return objectNode;
-        }
-
-        // Check for any "Print" and "Log" statements in a method
-        if (!(skipPrintStatements && skipLogStatements)) {
-            ArrayNode reasonObjectNode = mapper.createArrayNode();
-            boolean isFunctionSet = false;
-            List<InternalFeild> internalFields = methodMetaData.getInternallyRefferedFields();
-            for (InternalFeild internalField : internalFields) {
-                if (!skipPrintStatements && FieldUtil.containsPrintStatements(internalField)) {
-                    if (!isFunctionSet) {
-                        objectNode.put(JfuncConstants.ISFUNCTIONAL, false);
-                        isFunctionSet = true;
-                    }
-                    reasonObjectNode.add(FieldUtil.getReasonsForPrintStatements(internalField));
-                }
-                if (!skipLogStatements && FieldUtil.containsLogStatements(internalField)) {
-                    if (!isFunctionSet) {
-                        objectNode.put(JfuncConstants.ISFUNCTIONAL, false);
-                        isFunctionSet = true;
-                    }
-                    reasonObjectNode.add(FieldUtil.getReasonsForLogStatements(internalField));
-                }
-            }
-            if (!isFunctionSet) {
-                objectNode.put(JfuncConstants.ISFUNCTIONAL, true);
-            } else {
-                objectNode.set(JfuncConstants.REASONS, reasonObjectNode);
-            }
-        }
-        return objectNode;
-    }
+	private final String filePath;
+	private final Collection<File> classFiles;
+	private ClassReader classReader = null;
+	private ClassNode classNode;
+	private final List<FieldNode> fieldNodes = new ArrayList<FieldNode>();
+	private final AppMetadata appNode = new AppMetadata();
 
+	private List<ObjectNode> results = new ArrayList<ObjectNode>();
+
+	private final boolean skipLogStatements = false;
+	private final boolean skipPrintStatements = false;
+
+	public FunctionValidator(String path) throws FileNotFoundException {
```

Comment:
Can we extract out FunctionValidator. I mean according to me FunctionValidator should be specific to one class or one method. Our previous implementation is specific to one class. But now it is handling all the class files in a folder. Can we extract out this. I mean separation of concerns. 

Actually My plan was to keep this FunctionValidator as a Util class which does not care for which class and which method. This way we can keep our logic independent..

@pallaviroyal  wt do U say...